cypress-io/cypress
# fix: detect chrome browser process and tab crashes to no longer hang in CI

- Closes #6170
- Related to #22631

### User facing changelog
* When a chromium based browser tab or process crashes, Cypress will no longer hang indefinitely but close fail the current test and move on to the next.


### Additional details


We are detecting tab crashes when cdp tells us it crashed and process crashes when the process exits with specific code/signal combinations.

This affects run mode much more than open mode. In run mode we will move on to the next spec. In open mode we will re-launch the browser to the home screen if it crashes.

### Steps to test


I've created to specs that use CDP to crash the browser process and tab in the system tests e2e project. These work for any chromium based browser. This fix 'may' work for firefox but we have no way to test to see what signal/code combination we should be handling.

### How has the user experience changed?


Browser tab crash - before: process hangs indefinitely 

https://user-images.githubusercontent.com/1588747/197219147-305298dc-2673-4cb4-93f1-f4666558d5f0.mov

Browser tab crash - after: process fails current spec and moves on to the next.

https://user-images.githubusercontent.com/1588747/197216997-dee9f4d7-82bc-4109-b159-2c83779995ce.mov

Browser process crash - before: process hangs indefinitely 

https://user-images.githubusercontent.com/1588747/197219174-9d4a4e3d-cd77-43c1-adab-8e92a24583a8.mov

Browser process crash - after: process fails current spec and moves on to the next.

https://user-images.githubusercontent.com/1588747/197217877-8dd96604-d436-4ad5-89f5-28ee87278eeb.mov

### PR Tasks
